### ISSUE TITLE
Chipset des Adapters vor dem Kauf rausfinden

### ISSUE THREAD

**Issue by EliasKotlyar** [uholeschak/ediabaslib#39]
Hallo Herr Holeschak

Gibt es eine Möglichkeit vor dem Kauf eines "noName" ELM327 den Chipsatz rauszufinden?  Es geht im wesentlichen dadrum, das ich einen ELM Adapter mit BC-417 erwerben möchte. Pickit3 ist vorhanden und ich weiß wie man solche Devices flasht.
 
Auf den gängigen Seiten fand ich genau einen Händler, der eine "genauere" Beschreibung ink PCB-Bilder hatte. Ich habe mal seine Bilder kopiert - eventuell können sie mir ja sagen welcher Chipsatz was ist. Es gibt bei ihm 2 verschiedene Varianten. Eine mit grünem PCB, und eine mit rotem:

![grafik](https://user-images.githubusercontent.com/9529505/56100715-8402a680-5f1c-11e9-873b-0ace7109306c.png)
![grafik](https://user-images.githubusercontent.com/9529505/56100798-65e97600-5f1d-11e9-8aad-b21fa2fdaa8c.png)

![grafik](https://user-images.githubusercontent.com/9529505/56100801-700b7480-5f1d-11e9-945d-221cf16e075e.png)

![grafik](https://user-images.githubusercontent.com/9529505/56100804-74d02880-5f1d-11e9-84e2-af71c506b01b.png)

Ich gehe davon aus, das auf dem roten PCB ein HC05 Chip verbaut ist(dadrauf deutet die Markierung des Chips:m160309). Dieser sollte Pin sowie Software-Kompatibel zu HC04 sein. Wissen die Open-Source Firmware auch damit läuft? 

Vielen Dank im Voraus,
Grüße aus München
Elias

**Comment by uholeschak** [comment]
Hallo,
die Roten haben meisten die beste Qualität.

**Comment by nicolai1905** [comment]
please help me to flash elm327 v1.5 Bluetooth version 
https://github.com/uholeschak/ediabaslib/blob/master/docs/Replace_ELM327_HC04_Firmware.md

i dont understand how to flash . please help me to flash easy way . the elm327 i put on obd port my car after connecting to computer via software flash is possible ?? i need to replace firmware for coding my bmw . please help to replace this firmware easy way

**Comment by uholeschak** [comment]
Sorry there is no easy way and it depends from the adapter. This is for experts only

**Comment by ba114** [comment]
has anyone found a new adapter that shares the same header pins for programming?
If so, please share the link

**Comment by uholeschak** [comment]
The china adapters change every day. If you order another one you may get a different hardware.

**Comment by shar34** [comment]
I bought such an adapter (on 26.11.2019), not the same in tutorial, but it works fine.

link: https://aliexpress.ru/item/32893797457.html

![IMG_20191118_143810](https://user-images.githubusercontent.com/57906561/80961824-6e586b80-8e14-11ea-94b2-b15a80b65557.jpg)
![IMG_20191125_184915](https://user-images.githubusercontent.com/57906561/80961830-70bac580-8e14-11ea-930d-3f0dcfa2f4d2.jpg)

**Comment by ba114** [comment]
Is that the v1.5 product?

**Comment by shar34** [comment]
Yes, with PIC18F25K80.

**Comment by EliasKotlyar** [comment]
has someone managed to flash the HC05 Bluetooth Module? I mean not the PICKit releted part, but the onboard chip.

As far as i understand, the manufacturing process for this device is:
1. Order a OBD2 Plug with PIC18F25K80 which has some headers  to attach the PIcKIT programmer
2. Order a HC05 Bluetooth Chip 
3. Flash the OBD2 Plug with PicKIT
4. Remove the Bluetooth Chip from the module and replace it with HC05 Chip
5. Replace the firmware of the HC05 with CFW using a TTL-Adapter

I am stuck at point 5. I was doing  the following tutorial:
https://www.instructables.com/id/Upgrade-Your-3-Bluetooth-Module-to-Have-HID-Firmwa/
However, i am stuck with the bluesuite-software, which does not want to recognize the HC05. Does anyone encountered this problem? I am using 220Ohm Resistors.

**Comment by uholeschak** [comment]
Basically this is possible, but it's very advanced. I have used a solution with FTDI programmer:
https://github.com/lorf/csr-spi-ftdi/releases
http://www.instructables.com/id/AT-command-mode-of-HC-05-Bluetooth-module/

**Comment by ba114** [comment]
> I bought such an adapter (on 11/26/2019), not the same in tutorial, but it works fine.
> 
> link: https://aliexpress.ru/item/32893797457.html
> 
> ![IMG_20191118_143810](https://user-images.githubusercontent.com/57906561/80961824-6e586b80-8e14-11ea-94b2-b15a80b65557.jpg)
> ![IMG_20191125_184915](https://user-images.githubusercontent.com/57906561/80961830-70bac580-8e14-11ea-930d-3f0dcfa2f4d2.jpg)

can you please provide some more photos of the blue wire? also what resistor have you added in to the circuit?

**Comment by shar34** [comment]
I have no more photos, the blue wire soldered to the first chip leg (which is the MCLR pin), 512 resistor (http://kiloohm.info/smd3-resistor/512). All wires connected by this scheme, so if your board have no some or all test points, your may solder wires directly to legs:

![EK540967_4](https://user-images.githubusercontent.com/57906561/81051019-af5e8780-8ec9-11ea-8bf4-073444599111.JPG)

**Comment by ba114** [comment]
I bought 20 of these... any ideas how to hook them up for programming? they have Bk3432 chip.
Thanks

![Bluetooth](https://user-images.githubusercontent.com/21018877/81902944-4e8b2980-9604-11ea-886b-43c1bc719e6e.jpg)

**Comment by shar34** [comment]
My adapter have BK3432 too and it's working fine with default firmware.

**Comment by ba114** [comment]
> My adapter have BK3432 too and it's working fine with default firmware.

You didn't have to flash anything on this adapter?

**Comment by shar34** [comment]
> You didn't have to flash anything on this adapter?

Only flash PIC18F25K80 chip.

**Comment by ba114** [comment]
> > You didn't have to flash anything on this adapter?
> 
> Only flash PIC18F25K80 chip.

That's my issue, can't flash chip as it's got another board on top and no header pins

**Comment by denisrohr** [comment]
> > > You didn't have to flash anything on this adapter?
> > 
> > 
> > Only flash PIC18F25K80 chip.
> 
> That's my issue, can't flash chip as it's got another board on top and no header pins

I unsoldered top pcb, unsoldered the PIC using a heatgun, soldered the pins of the PIC directly to the pickit3, no resistors or capacitors and reprogrammed it using the "default" firmware. Resoldered it to the ELM and I could access the audi via deepobd. Attaching wires to the PIC while its on the pcb does NOT work, since there is some stuff on it that screws with the signal lines.

**Comment by OrpheeGT** [comment]
> 
> 
> > > > You didn't have to flash anything on this adapter?
> > > 
> > > 
> > > Only flash PIC18F25K80 chip.
> > 
> > 
> > That's my issue, can't flash chip as it's got another board on top and no header pins
> 
> I unsoldered top pcb, unsoldered the PIC using a heatgun, soldered the pins of the PIC directly to the pickit3, no resistors or capacitors and reprogrammed it using the "default" firmware. Resoldered it to the ELM and I could access the audi via deepobd. Attaching wires to the PIC while its on the pcb does NOT work, since there is some stuff on it that screws with the signal lines.

Hi, I have the same board, i unsoldered it :
![461-1024](https://user-images.githubusercontent.com/8038551/121504157-4c46fa80-c9d1-11eb-891d-163b3a99d242.jpg)

I would like to avoid unsolder the chip. 
Using multimeter : 
- Pin1 and Pin20 are wired and have a route to the full top right (currently unsoldered) pin.
- Pin8 (& pin9) is wired to Pin19

Lookiing @shar34 pic, could I do something like this or no way ? :
![pic](https://user-images.githubusercontent.com/8038551/121508266-f1170700-c9d4-11eb-8a9f-e8ca700548a2.png)


Brown links are currently wired pin (tested with multimeter)

**Comment by uholeschak** [comment]
The left front pin is the biggest Problem, but I have found to trick:

**Comment by OrpheeGT** [comment]
Hello,

I failed to program the module above, but I found the one described in original programming tutorial with RED PCB.

It has a YC1021 bluetooth chip.

I tried first to directly flash YC1021 version of FW, I can't connect to it with Deep OBD.
Then I flashed default FW and it worked.

But I guess I have then default ELM327 FW. I read Deep OBD is able to flash ELM327 FW, and I have the option to do so. But I don't see anywere how to choose YC1021 FW.

I tried to flash ELM327 FW with menu, it worked, and then after reboot Deep ODB suggested to flash a replacement FW for ELM327, so I did it.

It there a way with android to know the current ELM327 FW and see if it is the YC1021 modified one ?

Thanks

**Comment by uholeschak** [comment]
If you can't connect the YC1021 baud rate is configured incorrectly. The default FW uses the default baud rate.

**Comment by OrpheeGT** [comment]
There is no way to flash YC1021 ELM327 remotely from Deep ODB to enable 115200 speed ? I must solder on it as explained here ?

https://github.com/uholeschak/ediabaslib/blob/master/docs/Replace_Elm327_BT_Mini_Firmware.md

**Comment by uholeschak** [comment]
Correct.

**Comment by drbluetongue** [comment]
![IMG_20210803_214755](https://user-images.githubusercontent.com/12694883/127997048-a10bdf87-ad90-4eb4-a359-903f7ee651f0.jpg)

![IMG-20210803-WA0007](https://user-images.githubusercontent.com/12694883/127997288-3ed98504-3c75-48da-9b2d-162ce7e2bf5f.jpeg)


Hello!

Has anyone flashed one of these before? BK3231.

**Comment by drbluetongue** [comment]
So I found another adaptor that has pins on it. 

Flashed successfully the default file, but no Bluetooth afterwards :/

Avoid this BT chip
![IMG_20210810_190250](https://user-images.githubusercontent.com/12694883/128826803-1cb2483d-b669-426b-8253-743110bd9845.jpg)

**Comment by tdashmike** [comment]
@shar34 

Which firmware did you flash? It looks like the bluetooth chip on that one is a BP09497 but I cannot find any info on it. Thanks.

**Comment by shar34** [comment]
> @shar34
> 
> Which firmware did you flash? It looks like the bluetooth chip on that one is a BP09497 but I cannot find any info on it. Thanks.

Hello,

I flashed this firmware: **/CanAdapterElm/default/CanAdapterElm.X.production.unified.hex**, but I have BK3432 chip, not BP09497.

**Comment by tdashmike** [comment]
@shar34 Thanks, I tried that and it worked one time and it stopped working. I think I damaged the pads when I was trying to remove the board to see waht bluetooth moduel it was. I'll get a few more and possibly sell the extras on eBay.

**Comment by neuronet77** [comment]
I found this adapter in my "cable" collection. Do you know whether this one can be used? I can't see any "HC" chip on it :(

![20211230_094517](https://user-images.githubusercontent.com/4220037/147767580-90928673-f98e-4c66-b9db-6df521a165ea.jpg)

**Comment by uholeschak** [comment]
The Bluetooth chip seems to be somewhere else.

**Comment by neuronet77** [comment]
Sorry. had to pray it out to get to the back... :(
![20211230_101050](https://user-images.githubusercontent.com/4220037/147768890-f7c72686-c494-458a-87e1-fb74c563e72d.jpg)

**Comment by neuronet77** [comment]
I think I just got a "bingo" :D Now just waiting for the PICkit 3 programmer to be delivered... I saw a picture of you having connected 5 wires on the soldering dots on the one side of the board. It looks really like mine, so the wire colors you show there are the ones from the PICkit 3 cable?

Thanks so much for all you've done here! 

![20211230_113110](https://user-images.githubusercontent.com/4220037/147774959-e53dd846-d20e-4171-90d9-f7f9ac247c1a.jpg)
